pulsar-client-cpp/lib/ConsumerImpl.cc

```diff
@@ -826,6 +826,12 @@ Result ConsumerImpl::receiveHelper(Message& msg, int timeout) {
         messageProcessed(msg);
         return ResultOk;
     } else {
+        {
+            Lock lock(mutex_);
+            if (state_ != Ready) {
+                return ResultAlreadyClosed;
+            }
+        }
```

Comment:
```suggestion
        Lock lock(mutex_);
        if (state_ != Ready) {
            return ResultAlreadyClosed;
        }
```

It is already in the `else {}` block, there is no need to add another nested `{}` block.